tests/func/test_scm.py

```diff
@@ -95,8 +95,8 @@ def test_ignored(tmp_dir, scm):
     tmp_dir.gen({"dir1": {"file1.jpg": "cont", "file2.txt": "cont"}})
     tmp_dir.gen({".gitignore": "dir1/*.jpg"})
 
-    assert scm.is_ignored(tmp_dir / "dir1" / "file1.jpg")
-    assert not scm.is_ignored(tmp_dir / "dir1" / "file2.txt")
+    assert scm.is_ignored(str(tmp_dir / "dir1" / "file1.jpg"))
+    assert not scm.is_ignored(str(tmp_dir / "dir1" / "file2.txt"))
```

Comment:
After the typing update these `str()` calls should no longer be needed

Reply:
Also, it would be great if you could add a test case for the gitignore patterns that we were not handling correctly to this file

Reply:
You can generate an appropriate .gitignore the same way it's done in line 96

To get a .gitignore like:
```
dir/*
!dir/foo
```
you would generate it like:
```
tmp_dir.gen({".gitignore": "dir/*\n!dir/foo"})
```

Reply:
Done
Please LMK if sufficient

Thank you for the hint!